src/image/config.rs

```diff
@@ -0,0 +1,304 @@
+use std::{collections::HashMap, fs, path::Path};
+
+use anyhow::Result;
+use serde::ser::SerializeMap;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
```

Comment:
nit, unfortunately [`merge_imports`](https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#merge_imports) is still an unstable feature of rustfmt:
```suggestion
use serde::{Deserialize, Deserializer, Serialize, Serializer, ser::SerializeMap};
```